code_samples/simple_language/src/main/java/com/intellij/sdk/language/SimpleFoldingBuilder.java

```diff
@@ -0,0 +1,78 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+
+package com.intellij.sdk.language;
+
+import com.intellij.lang.ASTNode;
+import com.intellij.lang.folding.*;
+import com.intellij.openapi.editor.*;
+import com.intellij.openapi.project.*;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.sdk.language.psi.SimpleProperty;
+import org.jetbrains.annotations.*;
+
+import static com.intellij.sdk.language.SimpleAnnotator.*;
+
+import java.util.*;
+
+public class SimpleFoldingBuilder extends FoldingBuilderEx implements DumbAware {
+  @NotNull
+  @Override
+  public FoldingDescriptor[] buildFoldRegions(@NotNull PsiElement root, @NotNull Document document, boolean quick) {
+    // Initialize the group of folding regions that will expand/collapse together.
+    FoldingGroup group = FoldingGroup.newGroup(SIMPLE_PREFIX_STR);
+    // Initialize the list of folding regions
+    List< FoldingDescriptor > descriptors = new ArrayList< FoldingDescriptor >();
+    // Get a collection of the literal expressions in the document below root
+    Collection< PsiLiteralExpression > literalExpressions =
+          PsiTreeUtil.findChildrenOfType(root, PsiLiteralExpression.class);
+    // Evaluate the collection, using only Simple language literalExpressions
```

Comment:
what does "using only Simple language literalExpressions" mean?

Reply:
You're right, it's redundant.